lib/http/2/error.rb

```diff
@@ -25,7 +25,7 @@ class FlowControlError < ProtocolError; end
 
     # Raised on invalid stream processing: invalid frame type received or
     # sent, or invalid command issued.
-    class StreamError < ProtocolError; end
+    class InternalError < ProtocolError; end
```

Comment:
Why is this necessary? StreamError is a much better name indicating the actual type of error.